packages/swingset-liveslots/src/virtualReferences.js

```diff
@@ -133,18 +133,17 @@ export function makeVirtualReferenceManager(
     }
   }
 
-  function getFacetCount(baseRef) {
-    // Note that this only works if the VDO is in memory
-    const val = requiredValForSlot(baseRef);
-    if (Array.isArray(val)) {
-      return val.length;
+  function getFacetCount(kindID) {
+    const facetNames = kindInfoTable.get(`${kindID}`).facetNames;
```

Comment:
```suggestion
    const facetNames = getFacetNames(kindID);
```
I think it would also be nice to reorganize a bit for grouping these functions in better alignment with our conventions, e.g.
```js
  const kindInfoTable = new Map();

  …

  function getFacetNames(kindID) {
    return kindInfoTable.get(`${kindID}`).facetNames;
  }

  function getFacetCount(kindID) {
    const facetNames = getFacetNames(kindID);
    return facetNames ? facetNames.length : 1;
  }

  function getFacet(kindID, facets, facetIndex) {
    const facetNames = getFacetNames(kindID);
    const facetName = facetNames[facetIndex];
    facetName !== undefined || Fail`getFacet missing, ${kindID} [${facetIndex}]`;
    const facet = facets[facetName];
    facet || Fail`getFacet missing, ${kindID} [${facetIndex}] ${facetName}`;
    return facet;
  }
```

Reply:
I was aiming for a minimal diff, but yeah I'll move those functions to be closer to each other.